### PR TITLE
feat: Add RTL support to admin modal

### PR DESCRIPTION
**Pull request by sakhawy**
## Description

Added RTL support to cms-admin-modal

## Related resources

* #521

## Screenshots

Screenshot from 2024-03-18 11-10-57
Screenshot from 2024-03-18 11-11-03
Screenshot from 2024-03-21 13-33-02

## Checklist

* [ ] I have added or modified the tests when changing logic
* [X] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst)

### CODE REVIEW

**Comment by fsbraun** [comment]
@sakhawy Are the buttons and header positioned correctly?

**Comment by sakhawy** [comment]
Oh, I missed that! I'll adjust it, thanks!

**Comment by sakhawy** [comment]
@fsbraun Should I make this adjustment from ``djangocms-admin-style``? Or would it better be a ``django-cms`` PR since the styling of ``cms-modal-title`` is in ``django-cms/cms/static/cms/sass/components/_modal.scss``?

**Comment by fsbraun** [comment]
You're right, of course.

File: djangocms_admin_style/sass/components/_forms.scss [hunk 1]
```diff
@@ -305,7 +308,12 @@ form {
             float: left !important;
```
Comment (fsbraun) [inline djangocms_admin_style/sass/components/_forms.scss:308]: Couldn't that be `float: start`?

File: djangocms_admin_style/sass/components/_modal.scss [hunk 2]
```diff
@@ -66,12 +66,17 @@
                     float: left;
```
Comment (fsbraun) [inline djangocms_admin_style/sass/components/_modal.scss:66]:
```suggestion
                    float: start;
```

File: djangocms_admin_style/sass/components/_modal.scss [hunk 3]
```diff
@@ -143,15 +150,21 @@
             float: left;
```
Comment (fsbraun) [inline djangocms_admin_style/sass/components/_modal.scss:150]:
```suggestion
            float: start;
```

**Review by fsbraun — CHANGES_REQUESTED** [review]
I think you can use `float: start` and `float: end`...

**Comment by sakhawy** [comment]
@fsbraun ``inline-start`` and ``inline-end`` are not supported on relatively older browsers (as recent as 2023): https://caniuse.com/mdn-css_properties_float_inline-start so, I wanted to be extra careful. Should I make the change?

**Review by fsbraun — APPROVED** [review]

**Comment by joshyu** [comment]
@fsbraun , I think it's not a best practice to use too many `!important` in the CSS codes, which prevent us from customize those styes easily.

**Comment by fsbraun** [comment]
@joshyu I fully agree. djangocms-admin-style is not a good example for this, but has gone down that road may years ago. Alas, I believe with a bit of work this is fixable in most cases since all admin pages have the class `djangocms-admin-style` added to them allowing to override other styles by using higher specificity. It would be great if someone addressed this.

As a simpler work-around, I have created my own `djangocms-simple-admin-style` project (available on pypi): It tries to keep as many original django styles as possible and to void `!important`... Downside: It only works on browsers supporting CSS nesting (https://caniuse.com/css-nesting)

**Pull request merged**